### PR TITLE
Introduce grouping of logs

### PR DESCRIPTION
**Pull request by MariusStorhaug**
# Change

Use grouping of logs for [ADO](https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#formatting-commands) and [GH](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-5)

[Build Status](https://dev.azure.com/servicescode/infra-as-code-source/_build/latest?definitionId=2170&branchName=users%2Fmast%2FLogGrouping)

[AnalysisServices: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Review by MrMCake — CHANGES_REQUESTED** [review]
I'd undo the changes in ADO as it has task-wide grouping by default. Or is there another good reason to have it?

File: .github/actions/templates/publishModule/action.yml [hunk 1]
```diff
@@ -77,6 +85,7 @@ runs:
 
         # Publish the modified child resources
         foreach ($ModuleToUpdate in $ModulesToUpdate) {
+          Write-Output "::group::- $($ModuleToUpdate.TemplateFilePath)"
```
Comment (MrMCake) [inline .github/actions/templates/publishModule/action.yml:88]: I'd almost argue to just have a grouping at the beginning and the end of the tasks. I mimics ADO + it leaves the Logs to the script

File: .github/actions/templates/publishModule/action.yml [hunk 2]
```diff
@@ -111,6 +122,7 @@ runs:
...
```
Comment (MrMCake) [inline .github/actions/templates/publishModule/action.yml:125]: Same as comment above

File: .github/actions/templates/validateModuleDeployment/action.yml [hunk 3]
```diff
@@ -130,12 +144,17 @@ runs:
...
 
+        Write-Output "::endgroup::"
+
     # [Deployment execution] task(s)
     # ------------------------------
     - name: 'Deploy [${{ inputs.templateFilePath }}] with parameters [${{ inputs.parameterFilePath }}]'
       shell: pwsh
       id: deploy_step
       run: |
+        # Deploy [${{ inputs.templateFilePath }}] with parameters [${{ inputs.parameterFilePath }}]
```
Comment (MrMCake) [inline .github/actions/templates/validateModuleDeployment/action.yml:155]: Why a comment with tokens. It won't be displayed anyways. Should we maybe just have a generic `Grouping task` comment?

Comment (MrMCake) [reply .github/actions/templates/validateModuleDeployment/action.yml:155]: Some for the other added comments

File: .github/actions/templates/publishModule/action.yml (hunk 1 above)
Comment (MariusStorhaug) [reply .github/actions/templates/publishModule/action.yml:88]: It will not help us get to the specific outputs, but sure can do.

**Pull request merged**